src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -149,21 +152,36 @@ private function getRelationFieldsWithoutRelationType($relations, $include_neste
     }
 
     /**
-     * Changes the BelongsTo names in the input from request to allways
-     * have the foreign_key instead of the relation name.
-     * It only changes main relations not nested.
+     * Changes the BelongsTo names in the input from request to allways have the foreign_key instead of the relation name.
+     * When $belongs_to_fields are provided we will use those fields to replace relation names on the provided input.
      *
      * eg: user -> user_id
+     *
+     * @param  array  $input
+     * @param  array  $belongsToFields
+     * @return array
      */
-    private function changeBelongsToNamesFromRelationshipToForeignKey($input)
+    private function changeBelongsToNamesFromRelationshipToForeignKey($input, $belongs_to_fields = [])
     {
-        $belongs_to_fields = $this->getFieldsWithRelationType('BelongsTo');
+        if (empty($belongs_to_fields)) {
+            $belongs_to_fields = $this->getFieldsWithRelationType('BelongsTo');
+        } else {
+            foreach ($belongs_to_fields as $field) {
+                if (isset($field['subfields'])) {
+                    $belongs_to_fields = array_merge($field['subfields'], $belongs_to_fields);
+                }
+            }
+            $belongs_to_fields = array_filter($belongs_to_fields, function ($field) {
+                return isset($field['relation_type']) && $field['relation_type'] === 'BelongsTo';
+            });
```

Comment:
Wait wait wait:
- why are we getting `subfields` merged with `fields`? Won't this have negative effects?
- why are we re-doing the logic of `getFieldsWithRelationType()` here instead of calling `$this->getFieldsWithRelationType('BelongsTo', $fields)`?

I feel like this can be improved quite a bit. Let's talk about it.